core/src/main/java/hudson/tools/ToolDescriptor.java

```diff
@@ -54,6 +54,14 @@
 
     private T[] installations;
 
+    protected ToolDescriptor() {
+        super();
+    }
+
+    protected ToolDescriptor(Class<T> clazz) {
```

Comment:
Ah, probably yes, because the initial version of the split plugin will need to specify the core baseline prior to the split. So do it now. Should not be controversial.